### DISCUSSION TITLE
Error Downloading Canvas App Using PAC CLI: Remote Name Could Not Be Resolved

### DISCUSSION THREAD

**Discussion by FAC-1312** [microsoft/powerplatform-build-tools#740] [Q&A] [answered]
Hello everyone,

I'm encountering an issue while trying to download a canvas app using the Power Apps CLI (PAC). I followed the standard procedure to authenticate and download a canvas app from my Power Platform environment, but I keep receiving a DNS resolution error.

My auth profil is a service principal... The problem is encoutered only when I use my service principal profil. If i use my own connexion download is performed without problems. 

Command Attempted:
I executed the following command to download the canvas app:

`pac canvas download --name<App-ID> --extract-to-directory <Path-To-Save-The-App>`

Error Received:

Erreur : Une erreur s'est produite lors de l'envoi de la demande.
Le nom distant n'a pas pu être résolu: '79544f327b824b0296011990a48be3.05.environment.api.powerplatform.com'

Troubleshooting Steps Taken:
Checked for any typos in the environment URL and the app ID.
Made sure that I'm authenticated to the correct Power Platform environment.
Tried to access the Power Platform environment URL directly from a browser and donwloading my app  without encountering any issues 
Ensured that my PAC CLI is updated to the latest version.
Despite these efforts, the error persists, and I'm unable to proceed with downloading the canvas app.

Questions:
Has anyone faced a similar issue or does anyone know what might be causing this DNS resolution error?
Is there a specific format or consideration for the environment URL that I might be missing?
Any suggestions on additional troubleshooting steps I could take?
I appreciate any insights or assistance you can provide. Thank you in advance for your help!

**Comment by marcelbf** [comment, marked as answer]
Hi @FAC-1312 

Issue is with SPN indeed. You can't share a canvas app with an SPN - if you run pac canvas list using your SPN, do you see any app?

Any specific reason you have to use an SPN in your scenario? 
If you need to use SPN, you can wrap the msapp in a solution and use a solution to export it using your SPN auth.

Thanks for raising, we will clarify in the documentation that pac canvas list and download doesn't support SPN.
